src/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -255,11 +255,60 @@ public with sharing class AFFL_MultiRecordTypeMapper {
         for (SObjectField field : cf) {
             contactFieldStrings.add(field.getDescribe().getLabel());
         }
-        for(String primaryAfflFieldLabel : primaryAfflFieldLabels) {
+        for (String primaryAfflFieldLabel : primaryAfflFieldLabels) {
             if(contactFieldStrings.contains(primaryAfflFieldLabel)) {
                 validPrimaryFields.add(primaryAfflFieldLabel);
             }
         }
         return validPrimaryFields;
     }
+
+    /*******************************************************************************************************************************
+    * @description Returns the API name of the Primary Affiliation Lookup field from Affiliation_Type__c(Account Record Type Label)
+    * @param affiliationType The value of Affiliation_Type__c on Affiliation which is also related Account's Record Type Label Name
+    * @return lookupFieldAPIName The API name of the Primary Affiliation Lookup field
+    *********************************************************************************************************************************/
+    public String getPrimaryAffiliationLookupAPIFromAffiliationType(String affiliationType) {
+
+        if (affiliationType == null || affiliationType == '') {
+            return affiliationType;
+        }
+
+        String primayAfflLookupLabelOrAPIName = accRecTypeToContactLabel.get(affiliationType);
+
+        if (primayAfflLookupLabelOrAPIName == null) {
+            /* This means 'Account Record Type' in EDA settings is using API names for 'Account Record Type' in Affiliations mappings
+            and we are trying to find by label name, which is Affiliation_Type__c. So first get the API name of the Account Record type
+            from its label (affiliationType). Then use the record type's  API name to get the Contact's Priamry Affiliation's name
+            from settings which could be either API or label. */
+
+            String affiliationTypeAPI = accRecTypeLabelToAPI.get(affiliationType);
+            primayAfflLookupLabelOrAPIName = accRecTypeToContactLabel.get(affiliationTypeAPI);
+        }
+
```

Comment:
Let's add a guard clause here to exit if `primayAfflLookupLabelOrAPIName` is blank. We can't do anything at this point if it is.

Reply:
Updated per feedback. Resolving.